python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -742,8 +743,163 @@ def _plot_data(name, idx, marker_style=None):
+                    (t, a, e, c) for t, a, e, c in zip(p1_time, az_deg, el_deg, cn0_dbhz)]
+            figure.add_trace(go.Scatterpolargl(r=el_deg, theta=(90 - az_deg), text=text,
+                                               name=name, hoverinfo='name+text', hoverlabel={'namelength': -1},
+                                               mode='markers', marker=color_by_sv_format[-1]))
+            indices_by_system[system].append(len(figure.data) - 1)
+
+        # Add selection buttons for each system and for choosing between coloring by SV and C/N0.
+        num_traces = len(figure.data)
+        buttons = [dict(label='All', method='restyle', args=['visible', [True] * num_traces])]
+        for system, indices in sorted(indices_by_system.items()):
+            if len(indices) == 0:
+                continue
+            visible = np.full((num_traces,), False)
+            visible[indices] = True
+            buttons.append(dict(label=f'{str(system)} ({len(indices)})', method='restyle', args=['visible', visible]))
+        updatemenus = [{
+            'type': 'buttons',
+            'direction': 'left',
+            'buttons': buttons,
+            'x': 0.0,
+            'xanchor': 'left',
+            'y': 1.1,
+            'yanchor': 'top'
+        }]
+
+        updatemenus += [{
+            'type': 'buttons',
+            'direction': 'left',
+            'buttons': [
+                dict(label='Color By SV', method='restyle', args=['marker', color_by_sv_format]),
+                dict(label='Color By C/N0', method='restyle', args=['marker', color_by_cn0_format])
+            ],
+            'x': 0.0,
+            'xanchor': 'left',
+            'y': 1.045,
+            'yanchor': 'top'
+        }]
+
+        figure['layout']['updatemenus'] = updatemenus
+
+        self._add_figure(name='gnss_skyplot', figure=figure, title='GNSS Sky Plot')
+
+    def plot_gnss_cn0(self):
+        # The legacy GNSSSatelliteMessage contains data per satellite, not per signal. The plotted C/N0 values will
+        # reflect the L1 signal, unless L1 is not being tracked.
+        result = self.reader.read(message_types=[GNSSSatelliteMessage], **self.params)
+        data = result[GNSSSatelliteMessage.MESSAGE_TYPE]
+
+        if len(data.p1_time) == 0:
+            self.logger.info('No satellite data available. Skipping C/N0 plot.')
+            return
+
+        # Setup the figure.
+        figure = make_subplots(
+            rows=1, cols=1,  print_grid=False, shared_xaxes=True,
+            subplot_titles=['C/N0 (L1 Only)'])
+
+        figure['layout'].update(showlegend=True, modebar_add=['v1hovermode'])
+        figure['layout']['xaxis1'].update(title="Time (sec)", showticklabels=True)
+        figure['layout']['yaxis1'].update(title="C/N0 (dB-Hz)")
+
+        # Assign colors to each satellite.
+        data_by_sv = GNSSSatelliteMessage.group_by_sv(data)
+        svs = sorted(list(data_by_sv.keys()))
```

Comment:
Does this need to be a sorted list?

Reply:
Not right here technically, but I want to loop over it in order below so the traces on the plot are listed in order. I could sort it there but figured better to do it here just so `svs` is in order if we want to look at it for debugging.